### ISSUE TITLE
ExperientialMemory.query gives tiny positive neighbor scores a near-zero weight instead of 1.0

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#253]
## What's wrong

\`ExperientialMemory.query\` normalizes neighbor scores with:

\`\`\`python
neighbor_weights = positive_scores / jnp.maximum(score_sum, 1.0e-12)
\`\`\`

The \`1e-12\` floor is meant to guard the zero-score case (avoid dividing by zero when no neighbor is accepted), but it applies unconditionally rather than only when \`score_sum\` is exactly zero. Empirically, for a genuinely small-but-nonzero \`score_sum\` (e.g. a single accepted neighbor whose true similarity is exp(-36) ≈ 2.32e-16), the denominator floor produced a weight of approximately \`2.32e-4\` rather than \`1.0\` — the weights no longer summed to one for that neighbor, corrupting the weighted average over recovered actions.

Reproduced and verified via a real mutation-resistance transcript:

\`\`\`text
red (source fix reverted, regression test kept):
Mismatch at index [0]: 0.00023195227549877018 (ACTUAL), 1.0 (DESIRED)

green (fix restored):
47 passed
\`\`\`

## Reachability

\`ExperientialMemory\` is genuinely package-root exported (\`alberta_framework/__init__.py\` and \`alberta_framework/core/__init__.py\`, both import and \`__all__\`). \`ExperientialMemory.query\` is called by \`ExperientialMemoryPolicy.propose\`, which the public \`PrototypeAgent\`'s opt-in experiential-memory transaction calls via \`policy.propose\` — reachable through exported APIs with ordinary, configuration-valid inputs, not a hardcoded-safe-only path.

## Fix

Use the real \`score_sum\` as the denominator whenever it's positive, and fall back to \`1.0\` only when it's exactly zero (\`jnp.where(score_sum > 0.0, score_sum, 1.0)\`), rather than an unconditional epsilon floor via \`jnp.maximum\`.

PR incoming with a regression test using a real query at a configuration-valid sub-floor similarity, plus the mutation-resistance transcript.

**Comment by lalalune** [comment]
Maintainer review correction for the durable record: the regression is valid, but the narrative confuses the score with the resulting old weight. With key distance (6, 6) and distance_scale=1, squared_distance=36 and the accepted similarity/score is exp(-36), approximately 2.3195e-16. The old 1e-12 denominator floor turns that into a weight of approximately 2.3195e-4 instead of 1.0. A score_sum of 2.3e-4 would already divide by itself and was never affected. The actual defect window is 0 < score_sum < 1e-12; PR #254's test exercises that window and its source fix is mathematically correct.
